src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -2160,169 +2205,80 @@ int fmi3_xml_handle_ModelVariables(fmi3_xml_parser_context_t *context, const cha
         fmi3_xml_set_element_handle(context, "Tool",        FMI3_XML_ELM_ID(VariableTool));
     }
     else {
-         /* postprocess variable list */
-
+         // Post-process variable list
         fmi3_xml_model_description_t* md = context->modelDescription;
-        jm_vector(jm_voidp)* varByVR;
-        size_t i, numvar;
-
-        numvar = jm_vector_get_size(jm_named_ptr)(&md->variablesByName);
 
-        /* store the list of vars in original order */
-        {
-            size_t size = jm_vector_get_size(jm_named_ptr)(&md->variablesByName);
-            md->variablesOrigOrder = jm_vector_alloc(jm_voidp)(size,size,md->callbacks);
-            if (!md->variablesOrigOrder) {
-                fmi3_xml_parse_fatal(context, "Could not allocate memory");
-                return -1;
-            }
-            size_t i;
-            for(i= 0; i < size; ++i) {
-                jm_vector_set_item(jm_voidp)(md->variablesOrigOrder, i, jm_vector_get_item(jm_named_ptr)(&md->variablesByName,i).ptr);
+        // Create variablesByName
+        size_t nVars = jm_vector_get_size(jm_voidp)(&md->variablesOrigOrder);
+        jm_vector_resize(jm_named_ptr)(&md->variablesByName, nVars);  // Allocate a lot of memory to avoid resizing
+        jm_vector_resize(jm_named_ptr)(&md->variablesByName, 0);      // Make sure we start pushback at index=0
+        for (size_t i = 0; i < nVars; i++) {
+            jm_named_ptr named;
+            fmi3_xml_variable_t* v = jm_vector_get_item(jm_voidp)(&md->variablesOrigOrder, i);
+            named.name = v->name;
+            named.ptr = v;
+            jm_vector_push_back(jm_named_ptr)(&md->variablesByName, named);
+            if (v->aliases) {
+                fmi3_xml_alias_variable_t* alias;
+                for (size_t j = 0; j < jm_vector_get_size(jm_voidp)(v->aliases); j++) {
+                    alias = jm_vector_get_item(jm_voidp)(v->aliases, j);
+                    named.name = alias->name;
+                    named.ptr = v;
+                    jm_vector_push_back(jm_named_ptr)(&md->variablesByName, named);
+                }
             }
         }
-
-        /* sort the variables by names */
         jm_vector_qsort(jm_named_ptr)(&md->variablesByName, jm_compare_named);
```

Comment:
Not sure if we want to add this in this ticket, but once sorted, it would seem reasonably easy to add the global error check for unique names for variables & aliases.

Reply:
It actually already exists in `fmi3_check_variable_naming_conventions` but it's only called if FMIL is configured to check it. Unclear why we have that config option, or if there even exist more...

fmi_import_context.h
```
/**
    \brief If this configuration option is set, the model description will be
    checked to follow the variable naming conventions. Variables not following
    the convention will be logged.
*/
#define FMI_IMPORT_NAME_CHECK 1
```

```
    if (configuration & FMI3_XML_NAME_CHECK) {
        fmi3_check_variable_naming_conventions(md);
    }
```

This check is done after parsing all of modelDescription. I'll tried to move it to ModelVariables end-tag, but `configuration` variable isn't available there. Maybe not hard to fix, but don't want to do it right now. Adding FIXME instead.

Reply:
I'm okay with fixing this later on, but does this really qualify as naming convention? 
The issue I am seeing is that a valid modelDescription should not depend on the specific variables. If we allow identical names, then functions like get_variable_by_name will yield different results depending on how the variables were ordered.

Reply:
Right that would make it ambiguous and lead to secondary errors. Same in FMI 2.0 - unclear about FMI 1.0. The config value is opt-out, so the check is currently tested in `fmi3_xml_parsing_test.c`.

I'll add to cleanup list (OneNote) and see if we can remove that config, at least for FMI 3, and clarify that it will cause issues with FMI 2.